modules/ossm-installing-gateway-using-gateway-injection.adoc

```diff
@@ -0,0 +1,246 @@
+spec:
+  selector:
+    matchLabels:
+      istio: <gateway_name> 
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway <1>
+      labels:
+        istio: <gateway_name> <2>
+        sidecar.istio.io/inject: "true" <3>
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto <4>
+          securityContext:
+            capabilities:
+              drop:
+              - ALL
+            allowPrivilegeEscalation: false
+            privileged: false
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+          ports:
+          - containerPort: 15090
+            protocol: TCP
+            name: http-envoy-prom
+          resources:
+            limits:
+              cpu: 2000m
+              memory: 1024Mi
+            requests:
+              cpu: 100m
+              memory: 128Mi
+      securityContext:
+        sysctls:
+        - name: net.ipv4.ip_unprivileged_port_start
+          value: "0"
+      serviceAccountName: secret-reader <5>
+----
+<1> Indicates that the {istio} control plane uses the gateway injection template instead of the default sidecar template.
+<2> Ensure that a unique label is set for the gateway deployment. A label is required so that {istio} `Gateway` resources can select gateway workloads. 
+<3> Enables gateway injection by setting the `sidecar.istio.io/inject` parameter to `true`. If the name of the {istio} resource is not `default` you must use the `istio.io/rev: <istio_revision>` label, where the revision represents the active revision of the {istio} resource.
+<4> Sets the image field to `auto` so that the image automatically updates each time the pod starts.
+<5> Sets the `serviceAccountName` to the name of the `ServiceAccount` created in the previous step.
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f gateway-deployment.yml
+----
+
+. Verify that the gateway `Deployment` rollout was successful by running the following command:
++
+[source,terminal]
+----
+$ oc rollout status deployment/<gateway_name> -n <gateway_namespace>
+----
+
+. Create a YAML file named `gateway-service.yml` that contains the Kubernetes `Service` for the gateway. 
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <gateway_name>
+  namespace: <gateway_namespace>
+spec:
+  type: ClusterIP <1>
+  selector:
+    istio: <gateway_name> <2>
+  ports:
+    - name: status-port
+      port: 15021
+      protocol: TCP
+      targetPort: 15021
+    - name: http2
+      port: 80
+      protocol: TCP
+      targetPort: 80
+    - name: https
+      port: 443
+      protocol: TCP
+      targetPort: 443
+----
+<1> When you  set `spec.type` to `ClusterIP` the gateway `Service` is available to traffic within the cluster. If the gateway handles ingress traffic from outside the cluster, set `spec.type` to `LoadBalancer`. Alternatively, you can use OpenShift Routes. 
```

Comment:
It would be good to come back to this and a link after we have completed the "Get traffic into a mesh" section.

```suggestion
<1> When you  set `spec.type` to `ClusterIP` the gateway `Service` can only be accessed from within the cluster. If the gateway needs to handle ingress traffic from outside the cluster, set `spec.type` to `LoadBalancer`. Alternatively, you can use OpenShift Routes.
```

Reply:
done